app/controllers/application_controller.rb

```diff
@@ -1,7 +1,27 @@
 class ApplicationController < ActionController::Base
+  protect_from_forgery with: :null_session
+
+  def snowplow_domain_userid
+    # The JavaScript tracker sets first party cookies.
+    # It can be easier to model the data if the client-side and server-side
+    # trackers give the same domain_userid to their events.
+    # This method extracts the domain_userid from the cookie.
+    # The Ruby tracker does not automatically set a domain_userid.
+    sp_cookie = cookies.find { |key, _value| key =~ /^_sp_id/ }
+    sp_cookie.last.split(".").first if sp_cookie.present?
+  end
+
   # POST
   def track_page_view
+    # Here we are passing the domain_userid to the Snowplow object to set.
+    # This is not essential for the Ruby tracker functionality.
+    # Cookie information is only available within Controllers.
+
+    # Page title is a required argument for track page view.
+    # However, the server doesn't automatically have this information.
+    # This is one small reason why page views are easier to track client-side.
```

Comment:
It might be nice to add to this comment to list a positive, you can't be blocked by an ad blocker when doing this. So you get a better representation to actual number of page views. Also nice to compare client side and server side counts to see how many page views are going missing from ad blocking.